lib/mix/test/mix/tasks/format_test.exs

```diff
@@ -167,6 +167,20 @@ defmodule Mix.Tasks.FormatTest do
     assert output == ""
   end
 
+  test "checks that no error is raised with --check-formatted and --no-exit" do
+    capture_io("foo( )", fn ->
+      Mix.Tasks.Format.run(["--check-formatted", "--no-exit", "-"])
+    end)
+
+    message =
+      receive do
+        {:mix_shell, :info, [message]} ->
+          message
+      end
+
```

Comment:
You can still pattern match on the message, like this:

```suggestion
    assert_receive {:mix_shell, :info, [message]}
```

But ideally you would match on a prefix, something like:

```elixir
"foobar" <> _ = message
```

Reply:
Oh TIL! Thanks!